File: components/credentials/src/main/java/org/cloudfoundry/credhub/audit/entities/GetCredentialById.java
package org.cloudfoundry.credhub.audit.entities;

import org.cloudfoundry.credhub.audit.OperationDeviceAction;
import org.cloudfoundry.credhub.audit.RequestDetails;

public class GetCredentialById implements RequestDetails {
  private final String uuid;

  public GetCredentialById(final String uuid) {
    super();
    this.uuid = uuid;
  }

  @Override
  public OperationDeviceAction operation() {
    return OperationDeviceAction.GET;
  }

  public String getUuid() {
    return uuid;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/audit/entities/GetCredentialById.kt
package org.cloudfoundry.credhub.audit.entities

import org.cloudfoundry.credhub.audit.OperationDeviceAction
import org.cloudfoundry.credhub.audit.RequestDetails

class GetCredentialById(
    val uuid: String,
) : RequestDetails {
    override fun operation(): OperationDeviceAction = OperationDeviceAction.GET
}
